### ISSUE TITLE
Fix StreamView.hasNext()

### ISSUE THREAD

**Issue by zalokhan** [CorfuDB/CorfuDB#654]
hasNext returns an ambiguous result
```
    /** Returns whether or not there are potentially more entries in this
     * stream - this function may return true even if there are no entries
     * remaining, as addresses may have been acquired by other clients
     * but not written yet, or the addresses were hole-filled, or just failed.
     * @return      True, if there are potentially more entries in the stream.
     */
    boolean hasNext();
```
We need to either fix or remove this method.

**Comment by no2chem** [comment]
API, let's address this in v1.0-M2

**Comment by Maithem** [comment]
will be addressed by @annym work for this issue #1207
